src/newcharacter.cpp

```diff
@@ -2002,20 +2025,21 @@ tab_direction set_scenario(WINDOW *w, player *u, points_left &points)
                 cur_id = scens_length - 1;
             }
         } else if (action == "CONFIRM") {
-            u->start_location = sorted_scens[cur_id]->start_location();
-            u->str_max = 8;
-            u->dex_max = 8;
-            u->int_max = 8;
-            u->per_max = 8;
-            g->scen = sorted_scens[cur_id];
-            u->prof = &default_prof.obj();
-            u->empty_traits();
-            u->empty_skills();
-            u->add_traits();
-            points.init_from_options();
-            points.skill_points -= sorted_scens[cur_id]->point_cost();
-
-
+            if( sorted_scens[cur_id]->has_flag( "CITY_START" ) && !scenario_sorter.cities_enabled ) {}
```

Comment:
The styling is terrible, one can easily miss the `{}`. And the indentation is not needed, just `continue` when the selected scenario is not allowed:
```C++
if( confirm ) {
    if( scenario not allowed ) {
        continue;
    }
    ... // remaining original code.
```